ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/validation/RequestValidatorTest.java

```diff
@@ -0,0 +1,269 @@
+  private KsqlExecutionContext executionContext;
+
+  @Before
+  public void setUp() {
+    metaStore = new MetaStoreImpl(new InternalFunctionRegistry());
+    when(ksqlEngine.parse(any()))
+        .thenAnswer(inv -> new DefaultKsqlParser().parse(inv.getArgument(0)));
+    when(ksqlEngine.prepare(any()))
+        .thenAnswer(invocation ->
+            new DefaultKsqlParser().prepare(invocation.getArgument(0), metaStore));
+    executionContext = ksqlEngine;
+    serviceContext = TestServiceContext.create();
+    when(ksqlConfig.getInt(KsqlConfig.KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG))
+        .thenReturn(Integer.MAX_VALUE);
+    when(schemaInjector.forStatement(any())).thenAnswer(inv -> inv.getArgument(0));
+
+    final KsqlStream source = mock(KsqlStream.class);
+    when(source.getName()).thenReturn("SOURCE");
+    when(source.getSchema()).thenReturn(SCHEMA);
+
+    final KsqlStream sink = mock(KsqlStream.class);
+    when(sink.getName()).thenReturn("SINK");
+    when(sink.getSchema()).thenReturn(SCHEMA);
+
+    metaStore.putSource(source);
+    metaStore.putSource(sink);
+  }
+
+  @Test
+  public void shouldCallStatementValidator() {
+    // Given:
+    givenRequestValidator(
+        ImmutableMap.of(CreateStream.class, statementValidator)
+    );
+
+    final List<ParsedStatement> statements =
+        new DefaultKsqlParser().parse("CREATE STREAM x WITH (kafka_topic='x');");
+
+    // When:
+    validator.validate(statements, ImmutableMap.of(), "sql");
+
+    // Then:
+    verify(statementValidator, times(1)).validate(
+        argThat(is(preparedStatement(instanceOf(CreateStream.class)))),
+        eq(executionContext),
+        any(),
+        eq(ksqlConfig),
+        any()
+    );
+  }
+
+  @Test
+  public void shouldExecuteOnEngineIfNoCustomExecutor() {
+    // Given:
+    givenRequestValidator(
+        ImmutableMap.of()
+    );
+
+    final List<ParsedStatement> statements =
+        new DefaultKsqlParser().parse("SET 'property'='value';");
+
+    // When:
+    validator.validate(statements, ImmutableMap.of(), "sql");
+
+    // Then:
+    verify(ksqlEngine, times(1)).execute(
+        argThat(is(preparedStatement(instanceOf(SetProperty.class)))),
+        eq(ksqlConfig),
+        any());
+  }
+
+  @Test
+  public void shouldThrowExceptionIfValidationFails() {
+    // Given:
+    givenRequestValidator(
+        ImmutableMap.of(CreateStream.class, statementValidator)
+    );
+    doThrow(new KsqlException("Fail"))
+        .when(statementValidator).validate(any(), any(), any(), any(), any());
+
+    final List<ParsedStatement> statements =
+        new DefaultKsqlParser().parse("CREATE STREAM x WITH (kafka_topic='x');");
+
+    // Expect:
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage("Fail");
+
+    // When:
+    validator.validate(statements, ImmutableMap.of(), "sql");
+  }
+
+  @Test
+  public void shouldThrowIfNoValidatorAvailable() {
+    // Given:
+    givenRequestValidator(
```

Comment:
Just a suggestion, but you could have `givenRequestValidator(ImmutableMap.of());` in `setUp` so that you can exclude it from tests that don't need custom validators.